src/components/Swap/Summary/Details.tsx

```diff
@@ -27,61 +45,111 @@ interface DetailProps {
+export default function Details({ trade, slippage, gasUseEstimateUSD, inputUSDC, outputUSDC, impact }: DetailsProps) {
   const { inputAmount, outputAmount } = trade
   const inputCurrency = inputAmount.currency
   const outputCurrency = outputAmount.currency
   const integrator = window.location.hostname
   const feeOptions = useAtomValue(feeOptionsAtom)
+  const [exchangeRate] = useTradeExchangeRate({ trade })
 
-  const details = useMemo(() => {
+  const [details, estimationMessage] = useMemo(() => {
     const rows: Array<[string, string] | [string, string, Color | undefined]> = []
     // @TODO(ianlapham): Check that provider fee is even a valid list item
 
+    rows.push([t`Exchange rate`, exchangeRate])
+
     if (feeOptions) {
       const fee = outputAmount.multiply(feeOptions.fee)
       if (fee.greaterThan(0)) {
-        const parsedFee = formatCurrencyAmount({ amount: fee })
+        const parsedFee = formatCurrencyAmount(fee, NumberType.FiatGasPrice)
         rows.push([t`${integrator} fee`, `${parsedFee} ${outputCurrency.symbol || currencyId(outputCurrency)}`])
       }
     }
 
-    if (impact) {
-      rows.push([t`Price impact`, impact.toString(), impact.warning])
+    if (gasUseEstimateUSD) {
+      rows.push([t`Network fee`, `~${formatCurrencyAmount(gasUseEstimateUSD, NumberType.FiatGasPrice)}`])
     }
 
-    if (gasUseEstimateUSD) {
-      rows.push([t`Network fee`, `~${formatCurrencyAmount({ amount: gasUseEstimateUSD, isUsdPrice: true })}`])
+    if (impact) {
+      rows.push([t`Price impact`, impact.toString(), impact.warning])
     }
 
+    let estimationMessage = ''
     if (isExactInput(trade.tradeType)) {
-      const localizedMaxSent = formatCurrencyAmount({ amount: trade.minimumAmountOut(slippage.allowed) })
-      rows.push([t`Minimum received`, `${localizedMaxSent} ${outputCurrency.symbol}`])
+      const localizedMinReceived = formatCurrencyAmount(trade.minimumAmountOut(slippage.allowed), NumberType.TokenTx)
+      const minReceivedString = `${localizedMinReceived} ${outputCurrency.symbol}`
+      estimationMessage = t`Output is estimated. You will receive at least ${minReceivedString} or the transaction will revert.`
+      rows.push([t`Minimum output after slippage`, minReceivedString])
     } else {
-      const localizedMaxSent = formatCurrencyAmount({ amount: trade.maximumAmountIn(slippage.allowed) })
-      rows.push([t`Maximum sent`, `${localizedMaxSent} ${inputCurrency.symbol}`])
+      const localizedMaxSent = formatCurrencyAmount(trade.maximumAmountIn(slippage.allowed), NumberType.TokenTx)
+      const maxSentString = `${localizedMaxSent} ${inputCurrency.symbol}`
+      estimationMessage = t`Output is estimated. You will send at most ${maxSentString} or the transaction will revert.`
+      rows.push([t`Maximum sent`, maxSentString])
     }
 
-    rows.push([t`Slippage tolerance`, `${slippage.allowed.toFixed(2)}%`, slippage.warning])
-
-    return rows
+    return [rows, estimationMessage]
```

Comment:
nit: I find it clearer to return an object, so you can destructure and retain the names; vs an array, where names can change when destructuring and similar types may be accidentally swapped.